src/driver-steelseries.c

```diff
@@ -443,9 +443,29 @@ steelseries_write_report_rate(struct ratbag_profile *profile)
 	uint8_t buf[STEELSERIES_REPORT_SIZE] = {0};
 
 	if (device_version == 1) {
+		char reported_rate;
+		switch (profile->hz) {
+			case 125:
+				reported_rate = 0x04;
+				break;
+			case 250:
+				reported_rate = 0x03;
+				break;
+			case 500:
+				reported_rate = 0x02;
+				break;
+			case 1000:
+				reported_rate = 0x01;
+				break;
+			default:
+				log_info(device->ratbag, "invalid report rate (%d); falling back to 1000 hz.\n", profile->hz);
```

Comment:
We should fallback to the closest here.

Reply:
I'm not really sure how to implement that... What about this? It falls back not to the closest one, but to the higher one though.
```c
static unsigned int REPORT_RATES_V1[] = { 125, 250, 500, 1000 };
static uint8_t REPORT_RATES_V1_ENCODED[] = { 0x04, 0x03, 0x02, 0x01};

...

char reported_rate = '\0';
for (size_t i = 0; i < ARRAY_LENGTH(REPORT_RATES_V1); i++) {
	if (profile->hz <= REPORT_RATES_V1[i]) {
		reported_rate = REPORT_RATES_V1_ENCODED[i];
		profile->hz = REPORT_RATES_V1[i];
		break;
	};
}
`

Reply:
Done that and rebased to master.